### ISSUE TITLE
Set up a homebrew package.

### ISSUE THREAD

**Issue by cube2222** [opentofu/opentofu#14]
We can start with a homebrew tap initially (as we can prepare that easily in private).

See
- https://github.com/spacelift-io/spacectl/blob/main/.github/workflows/release.yml
- https://github.com/spacelift-io/homebrew-spacelift
for inspiration.

Then, after the first public stable release, we can also contribute a homebrew-core package. See https://github.com/Homebrew/homebrew-core/blob/master/Formula/t/terraform.rb for inspiration.

The brew should be `brew install opentofu`

**Comment by thschue** [comment]
Could take this over (if it's ok to do this with goreleaser). Would need an additional repo (for brew) and a PAT in this repo which could push to the brew repo.

**Comment by Yantrio** [comment]
Hi @thschue,

First off, thank you so much for your proactive approach and for looking to take this task onboard. We genuinely appreciate your efforts and dedication towards contributing.

That said, will be renaming the project soon. Because of this, it might be more efficient to hold off on the homebrew setup until after these changes are implemented to ensure consistency and avoid any potential confusion. Furthermore, we're expecting our alpha release in approximately 2-3 weeks, which seems like an opportune time to align these tasks.

Would you be open to assisting with the homebrew package setup after the alpha release once the renaming process is settled? This way, we can make sure everything is streamlined and seamless for our users.

Thank you for your understanding and patience. We're looking forward to collaborating closely with you on this!

**Comment by thschue** [comment]
Hello and thank you! For me, it's also ok if I could add this later :-)

**Comment by Hefeweizen** [comment]
[GoReleaser](https://goreleaser.com/) is in use.  It offers [homebrew support](https://goreleaser.com/customization/homebrew/).  This should be relatively straightforward to set up, but two details immediately spring to mind:

1. We'll need a repo, likely under the github org for opentffoundation, for this.  I'm unsure who to coordinate this with.  @thschue worked on https://github.com/opentffoundation/opentf/pull/362 which is related work, so maybe he can point us in a good direction?
2. It looks like only the commercial version of GoReleaser supports [versioned tap formulas](https://goreleaser.com/customization/homebrew/#versioned-formulas) (e.g. opentf@0.1 vs opentf@v0.2).  TF was rather version sensitive, so I expect that to be an ongoing issue; I expect formula versioning to be extremely useful.  Is a licensed version of GoReleaser already in use?  [There's no](https://goreleaser.com/pro/#pricing-sponsors) [mention](https://beckersoft.gumroad.com/l/goreleaser) of what accommodations can be made for open source projects, if any. 

To be clear, I don't think #2 should be a blocker on implementation, just that it's an additional issue that will come up quickly.

**Comment by lfacchinelli** [comment]
> Would need an additional repo (for brew) and a PAT in this repo which could push to the brew repo.

Is a new repo needed just for tap? I thought we can just drop homebrews' config on this repo and it would work by adding as a new tap:
``` 
brew tap opentofu/opentofu

**Comment by cube2222** [comment]
Alright, I've created the https://github.com/opentofu/homebrew-opentofu repo for this.

@thschue would you be willing to work on this in the near future?

I believe it would make sense to have separate taps for the alpha and stable releases? Something like `opentofu/opentofu@alpha`? Not sure how hard that would be to accomplish with goreleaser.

**Comment by thschue** [comment]
I think I can do this in the next days. We will also need a PAT which is able to push in the brew repo.

TBH i'm not sure if it works this way, I will investigate a bit. I think by default, it will take the release tag, but I'm not sure how brew behaves with alpha/beta releases in the tags.

**Comment by cube2222** [comment]
@thschue Homebrew is evergreen, it only contains the "latest" version.

In effect that means, that we'll want a separate formula for alpha and for stable. There's a naming convention where stable is just the pure name, and older or prerelease versions get a `@channel_name`: https://docs.brew.sh/Versions

You can see how it works for gcc here: https://github.com/Homebrew/homebrew-core/tree/master/Formula/g

In other words, there'd be two formulas: opentofu, and opentofu@alpha

The links in the issue description should also be a good inspiration.

**Comment by RoseSecurity** [comment]
Are we planning on using`Goreleaser` in https://github.com/opentofu/homebrew-opentofu? If you need help writing any documentation, feel free to reach out.

**Comment by cube2222** [comment]
Yup, goreleaser is already used in this repo, and will be used to push a release to the homebrew tap.

**Comment by casualuser** [comment]
> Alright, I've created the https://github.com/opentofu/homebrew-opentofu repo for this.
> 
> @thschue would you be willing to work on this in the near future?
> 
> I believe it would make sense to have separate taps for the alpha and stable releases? Something like `opentofu/opentofu@alpha`? Not sure how hard that would be to accomplish with goreleaser.

looking for any content of `opentofu/opentofu` to appear to make a first tests at my side

**Comment by cube2222** [comment]
Since Homebrew has [accepted OpenTofu into the Core repo](https://github.com/Homebrew/homebrew-core/pull/149678), I believe we don't need this tap anymore and can close the issue?

cc @thschue
